diff --git a/pkg/v8/chartconfig/chartconfig.go b/pkg/v8/chartconfig/chartconfig.go
index 3f880013..a23ca2f5 100644
--- a/pkg/v8/chartconfig/chartconfig.go
+++ b/pkg/v8/chartconfig/chartconfig.go
@@ -76,7 +76,8 @@ func (c *ChartConfig) newTenantK8sClient(ctx context.Context, clusterConfig Clus
 
 func containsChartConfig(list []*v1alpha1.ChartConfig, item *v1alpha1.ChartConfig) bool {
 	for _, l := range list {
-		if reflect.DeepEqual(item, l) {
+		if reflect.DeepEqual(item.Spec, l.Spec) &&
+			reflect.DeepEqual(item.Labels, l.Labels) {
 			return true
 		}
 	}
